test/extended/router/http2.go

```diff
@@ -54,29 +61,77 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 	defer g.GinkgoRecover()
 
 	var (
-		configPath = exutil.FixturePath("testdata", "router", "router-http2.yaml")
-		oc         = exutil.NewCLI("router-http2")
+		http2ServiceConfigPath     = exutil.FixturePath("testdata", "router", "router-http2.yaml")
+		http2RoutesConfigPath      = exutil.FixturePath("testdata", "router", "router-http2-routes.yaml")
+		http2RouterShardConfigPath = exutil.FixturePath("testdata", "router", "router-shard.yaml")
+		http2SourceBackendPath     = exutil.FixturePath("testdata", "router", "router-http2-server.backend")
+
+		oc = exutil.NewCLI("router-http2")
+
+		shardConfigPath string // computed
 	)
 
 	// this hook must be registered before the framework namespace teardown
 	// hook
 	g.AfterEach(func() {
 		if g.CurrentGinkgoTestDescription().Failed {
-			client := routeclientset.NewForConfigOrDie(oc.AdminConfig()).RouteV1().Routes(oc.KubeFramework().Namespace.Name)
-			if routes, _ := client.List(context.Background(), metav1.ListOptions{}); routes != nil {
-				outputIngress(routes.Items...)
-			}
 			exutil.DumpPodLogsStartingWith("http2", oc)
+			exutil.DumpPodLogsStartingWithInNamespace("router", "openshift-ingress", oc.AsAdmin())
+		}
+		if len(shardConfigPath) > 0 {
+			oc.AsAdmin().Run("delete").Args("-n", "openshift-ingress-operator", "-f", shardConfigPath).Execute()
```

Comment:
Should we print an error when deleting here just in case (might be useful when running this test outside of CI).

Reply:
Done in 1506505c801a8bd5a37195527b45bf70370a2a93. Thanks.